src/main/java/Orca/TaskManager.java

```diff
@@ -0,0 +1,251 @@
+            }
+        }
+        System.out.println(Orca.LINE);
+    }
+
+    public void addTodoTask(String taskDescription) {
+        try {
+            taskDescription = taskDescription.trim();
+            // Ensure there is a description after the command.
+            if (!taskDescription.contains(" ")) {
+                throw new IllegalArgumentException("The description of a todo cannot be empty.");
+            }
+            String description = taskDescription.substring(taskDescription.indexOf(" ") + 1).trim();
+            if (description.isEmpty()) {
+                throw new IllegalArgumentException("The description of a todo cannot be empty.");
+            }
+            Todo newEntry = new Todo(description);
+            tasks.add(newEntry);
+            saveTasksToStorage();
+            printAddedTodoTask(newEntry);
+        } catch (Exception e) {
+            System.out.println(Orca.LINE + "\nError adding todo task: " + e.getMessage() + "\n" + Orca.LINE);
+        }
+    }
+
+    public void addDeadlineTask(String taskDescription) {
+        try {
+            taskDescription = taskDescription.trim();
+            if (!taskDescription.contains(" ")) {
+                throw new IllegalArgumentException("The description of a deadline cannot be empty.");
+            }
+            String[] parts = taskDescription.split("/", -1);
+            if (parts.length != 2) {
+                throw new IllegalArgumentException(
+                        "A deadline task should contain exactly one '/' separator. " +
+                                "Format: deadline <description> /by <time>."
+                );
+            }
+            String command = "deadline";
+            if (!parts[0].toLowerCase().startsWith(command)) {
+                throw new IllegalArgumentException("Deadline task must start with 'deadline'.");
+            }
+            String description = parts[0].substring(command.length()).trim();
+            if (description.isEmpty()) {
+                throw new IllegalArgumentException("The description of a deadline cannot be empty.");
+            }
+            String byPart = parts[1].trim();
+            if (!byPart.toLowerCase().startsWith("by")) {
+                throw new IllegalArgumentException("Deadline task should specify '/by <time>'.");
+            }
+            String deadlineTime = byPart.substring("by".length()).trim();
+            if (deadlineTime.isEmpty()) {
+                throw new IllegalArgumentException("The deadline time cannot be empty.");
+            }
+            Deadline newEntry = new Deadline(taskDescription);
+            tasks.add(newEntry);
+            saveTasksToStorage();
+            printAddedDeadlineTask(newEntry, deadlineTime);
+        } catch (Exception e) {
+            System.out.println(Orca.LINE + "\nError adding deadline task: " + e.getMessage() + "\n" + Orca.LINE);
+        }
+    }
+
+    public void addEventTask(String taskDescription) {
```

Comment:
try to keep every method to less than equals to 30 lines of code. Avoid very long methods